crates/matrix-sdk-crypto/src/verification/requests.rs

```diff
@@ -634,7 +648,23 @@ impl VerificationRequest {
                 if let Some((sas, content)) =
                     s.clone().start_sas(self.we_started, self.inner.clone().into()).await?
                 {
-                    self.verification_cache.insert_sas(sas.clone());
+                    // We may have previously started QR verification and generated a QR code. If we
+                    // now switch to SAS flow, the previous verification has to be replaced
+                    if cfg!(feature = "qrcode") {
```

Comment:
Hmm, combining `cfg!()` with the `#[cfg()]` block seems a bit unidiomatic.

Is there something wrong with replacing the else here with just;
```rust
#[cfg(not(feature = "qrcode"))]
self.verification_cache.insert_sas(sas.clone());
```

Reply:
Ah, I think you're looking for the `cfg_if!()` macro then: https://github.com/alexcrichton/cfg-if. Though not sure if we want to start depending on that.

Reply:
If we have more than a single place where it would be useful, I'm not against it. It's a simple `macro_rules!` macro crate, should compile instantly pretty much.

Reply:
Conceptually yes, this is what I had in mind. Looking at `Cargo.lock` we already seem to have `cfg-if` as transitive dependency so perhaps adding it explicitely does not actually add much overhead?

Reply:
Alright, If you prefer `cfg-if` feel free to add it and use it here.